src/components/MultiHopTrade/hooks/useTradeExecution/useTradeExecution.ts

```diff
@@ -95,18 +107,59 @@ export const useTradeExecution = ({
       })
 
       // execute the trade and attach then cancel callback
-      cancelPollingRef.current = await execution.exec({
-        swapperName,
-        tradeQuote,
-        stepIndex: activeStepOrDefault,
-        accountMetadata,
-        quoteSellAssetAccountId: sellAssetAccountId,
-        quoteBuyAssetAccountId: buyAssetAccountId,
-        wallet,
-        supportsEIP1559,
-        slippageTolerancePercentageDecimal,
-        getState: store.getState,
-      })
+      if (execution.exec) {
+        const output = await execution.exec?.({
+          swapperName,
+          tradeQuote,
+          stepIndex: activeStepOrDefault,
+          accountMetadata,
+          quoteSellAssetAccountId: sellAssetAccountId,
+          quoteBuyAssetAccountId: buyAssetAccountId,
+          wallet,
+          supportsEIP1559,
+          slippageTolerancePercentageDecimal,
+          getState: store.getState,
+        })
+
+        cancelPollingRef.current = output?.cancelPolling
+      }
+
+      if (execution.exec2) {
+        const accountNumber = accountMetadata.bip44Params.accountNumber
+        const chainId = tradeQuote.steps[activeStepOrDefault].sellAsset.chainId
+        const adapter = assertGetEvmChainAdapter(chainId)
+        const from = await adapter.getAddress({ accountNumber, wallet })
+        const account = await adapter.getAccount(from)
+        const evm: EvmTradeExecutionProps = {
+          from,
+          nonce: numberToHex(account.chainSpecific.nonce),
```

Comment:
note nonce being parsed here and passed into callback instead of inside chain adapter which would require coupling everything to HDWallet (like previously)